### PR TITLE
APP MANAGER V2 Split part 002

### PR DESCRIPTION
**Pull request by biyeun**
@benrudolph @orangejenny 
cc @dannyroberts

### CODE REVIEW

File: corehq/apps/app_manager/suite_xml/post_process/workflow.py [hunk 1]
```diff
@@ -370,7 +370,7 @@ def add_datums_for_target(module, source_form_dm, allow_missing=False):
                     elif not allow_missing:
                         raise SuiteValidationError(
                             u"The '{}' module is not properly configured to have a Case List Registration Form. "
-                            u"All forms in the module should have the same case management configuration.".format(
+                            u"All forms in the case list should have the same case management configuration.".format(
```
Comment (dimagimon) [inline corehq/apps/app_manager/suite_xml/post_process/workflow.py]: E501 line too long (117 > 115 characters)

File: corehq/apps/app_manager/views/apps.py [hunk 2]
```diff
@@ -120,9 +120,13 @@ def default_new_app(request, domain):
...
     app = Application.new_app(domain, _("Untitled Application"), lang=lang)
-    module = Module.new_module(_("Untitled Module"), lang)
-    app.add_module(module)
-    form = app.new_form(0, "Untitled Form", lang)
+
+    if not toggles.APP_MANAGER_V2.enabled(domain):
+        # APP MANAGER V2 is completely blank on new app
+        module = Module.new_module(_("Untitled Module"), lang)
+        app.add_module(module)
+        form = app.new_form(0, "Untitled Form", lang)
```
Comment (dimagimon) [inline corehq/apps/app_manager/views/apps.py]: F841 local variable 'form' is assigned to but never used

File: corehq/apps/app_manager/views/modules.py [hunk 3]
```diff
@@ -818,27 +822,46 @@ def new_module(request, domain, app_id):
...
+                    register.source = f.read()
+                register.actions.open_case = OpenCaseAction(
+                    condition=FormActionCondition(type='always'),
+                    name_path=u'/data/name')
+                register.actions.update_case = UpdateCaseAction(
+                    condition=FormActionCondition(type='always'))
+
+                # make case type unique across app
+                app_case_types = set(
+                    [module.case_type for module in app.modules if
```
Comment (dimagimon) [inline corehq/apps/app_manager/views/modules.py]: F812 list comprehension redefines 'module' from line 829

File: corehq/apps/app_manager/views/modules.py [hunk 4]
```diff
@@ -818,27 +822,46 @@ def new_module(request, domain, app_id):
...
+                     module.case_type])
+                module.case_type = 'case'
+                suffix = 0
+                while module.case_type in app_case_types:
+                    suffix = suffix + 1
+                    module.case_type = 'case-{}'.format(suffix)
+            else:
+                form = app.new_form(module_id, "Survey", lang)
```
Comment (dimagimon) [inline corehq/apps/app_manager/views/modules.py]: F841 local variable 'form' is assigned to but never used

File: corehq/apps/app_manager/views/apps.py (hunk 2 above)
Comment (biyeun) [inline corehq/apps/app_manager/views/apps.py]: ignore

File: corehq/apps/app_manager/views/modules.py (hunk 4 above)
Comment (biyeun) [inline corehq/apps/app_manager/views/modules.py]: ignore

File: corehq/apps/app_manager/views/apps.py (hunk 2 above)
Comment (dimagimon) [inline corehq/apps/app_manager/views/apps.py]: F841 local variable 'form' is assigned to but never used

File: corehq/apps/app_manager/views/modules.py (hunk 4 above)
Comment (dimagimon) [inline corehq/apps/app_manager/views/modules.py]: F841 local variable 'form' is assigned to but never used

File: corehq/apps/app_manager/views/apps.py (hunk 2 above)
Comment (dimagimon) [inline corehq/apps/app_manager/views/apps.py]: F841 local variable 'form' is assigned to but never used

File: corehq/apps/app_manager/views/modules.py (hunk 3 above)
Comment (dimagimon) [inline corehq/apps/app_manager/views/modules.py]: F812 list comprehension redefines 'module' from line 829

File: corehq/apps/app_manager/views/modules.py (hunk 4 above)
Comment (dimagimon) [inline corehq/apps/app_manager/views/modules.py]: F841 local variable 'form' is assigned to but never used

**Comment by biyeun** [comment]
@benrudolph / @orangejenny / @dannyroberts ✅ !!

**Comment by benrudolph** [comment]
👏 nice

fairly hard for me to wrap my head around all the changes, but did the best i could

**Pull request merged**